### PR TITLE
Enhanced deployment tracking

### PR DESCRIPTION
**Pull request by jnummelin**
This PR enhances the service deployment tracking to catch also errors. This way the deployment waiting on the cli side can be terminated and proper error shown to users.

fixes (partially) #1275

### CODE REVIEW

**Comment by jakolehm** [comment]
@jnummelin @nevalla I think we should include this to 0.17?

**Comment by jnummelin** [comment]
At least the related issue #1275 is tagged for 0.17

**Comment by nevalla** [comment]
Yes we should

File: server/app/models/grid_service_deploy.rb [hunk 1]
```diff
@@ -1,10 +1,15 @@
 class GridServiceDeploy
   include Mongoid::Document
   include Mongoid::Timestamps
+  include Mongoid::Enum
+
```
Comment (jakolehm) [inline server/app/models/grid_service_deploy.rb]: Too many empty lines.

**Pull request merged**